File: app/src/main/java/com/intoverflown/sasakazi/ui/discussions/mobile/MessageViewHolder.java
package com.intoverflown.sasakazi.ui.discussions.mobile;

import android.net.Uri;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import com.intoverflown.sasakazi.R;
import de.hdodenhof.circleimageview.CircleImageView;

public class MessageViewHolder extends RecyclerView.ViewHolder {

    private static final String TAG = "MessageViewHolder";

    TextView messageTextView;
    ImageView messageImageView;
    TextView messengerTextView;
    CircleImageView messengerImageView;

    public MessageViewHolder(View v) {
        super(v);
        messageTextView = (TextView) itemView.findViewById(R.id.messageTextView);
        messageImageView = (ImageView) itemView.findViewById(R.id.messageImageView);
        messengerTextView = (TextView) itemView.findViewById(R.id.messengerTextView);
        messengerImageView = (CircleImageView) itemView.findViewById(R.id.messengerImageView);
    }

    public void bindMessage(Messages messages) {
        if (messages.getText() != null) {
            messageTextView.setText(messages.getText());
            messengerTextView.setText(messages.getFullname());
            messageTextView.setVisibility(TextView.VISIBLE);
            messageImageView.setVisibility(ImageView.GONE);
        } else if (messages.getImageUrl() != null) {
            String imageUrl = messages.getImageUrl();
            if (imageUrl.startsWith("gs://")) {
                StorageReference storageReference = FirebaseStorage.getInstance()
                        .getReferenceFromUrl(imageUrl);

                storageReference.getDownloadUrl()
                        .addOnSuccessListener(new OnSuccessListener<Uri>() {
                            @Override
                            public void onSuccess(Uri uri) {
                                String downloadUrl = uri.toString();
                                Glide.with(messageImageView.getContext())
                                        .load(downloadUrl)
                                        .into(messageImageView);
                            }
                        })
                        .addOnFailureListener(new OnFailureListener() {
                            @Override
                            public void onFailure(@NonNull Exception e) {
                                Log.w(TAG, "Getting download url was not successful.", e);
                            }
                        });
            } else {
                Glide.with(messageImageView.getContext())
                        .load(messages.getImageUrl())
                        .into(messageImageView);
            }

            messageImageView.setVisibility(ImageView.VISIBLE);
            messageTextView.setVisibility(TextView.GONE);
        }
    }
}

File: app/src/main/java/com/intoverflown/sasakazi/ui/discussions/mobile/MessageViewHolder.kt
package com.intoverflown.sasakazi.ui.discussions.mobile


import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.intoverflown.sasakazi.R


class MessageViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val TAG = "MessageViewHolder"

    var messageReceiverTextView: TextView? = null
    var messageSenderTextView: TextView? = null

    fun MessageViewHolder(v: View?) {
        super.itemView
        messageReceiverTextView = itemView.findViewById<View>(R.id.textViewReceiver) as TextView
        messageSenderTextView = itemView.findViewById<View>(R.id.textViewSender) as TextView
    }

    fun bindMessage(mobileMessages: MobMessages) {
        if (mobileMessages.getText() != null) {
            messageReceiverTextView!!.setText(mobileMessages.getText())
            messageSenderTextView!!.setText(mobileMessages.getName())
            messageReceiverTextView!!.setVisibility(TextView.VISIBLE)
        }
    }
}